File: web/src/types/report.ts
export interface ReportMetadata {
  id: number;
  report_type: string;
  report_date: string;
  pool_id?: number;
  template_id?: number;
  status: 'pending' | 'running' | 'done' | 'failed';
  format: 'html' | 'markdown' | 'json';
  file_path?: string;
  file_size?: number;
  error_msg?: string;
  started_at?: string;
  finished_at?: string;
  created_at: string;
}

export interface ReportGenerateRequest {
  pool_id: number;
  report_type: string;
  format: 'html' | 'markdown' | 'json';
  template_id?: number;
}

export interface ReportStatus {
  id: number;
  status: 'pending' | 'running' | 'done' | 'failed';
  progress?: number;
  error_msg?: string;
}

File: app/schemas/report.py
"""Report generation Pydantic schemas.

Provides request/response models for report generation, status tracking,
and report listing operations.
"""

from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


# ------------------------------------------------------------------
# Report generation request
# ------------------------------------------------------------------

class ReportGenerateRequest(BaseModel):
    """Request model for triggering report generation."""

    report_type: str = "pool_weekly"
    pool_id: Optional[int] = None
    format: str = "html"
    template_id: Optional[int] = None


# ------------------------------------------------------------------
# Report response models
# ------------------------------------------------------------------

class ReportResponse(BaseModel):
    """Response model for a generated report."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    report_type: str
    report_date: date
    pool_id: Optional[int] = None
    status: str
    format: str
    file_path: Optional[str] = None
    file_size: Optional[int] = None
    created_at: Optional[datetime] = None


class ReportStatusResponse(BaseModel):
    """Response model for report generation status."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    status: str
    file_path: Optional[str] = None
    file_size: Optional[int] = None
    error_msg: Optional[str] = None
    started_at: Optional[datetime] = None
    finished_at: Optional[datetime] = None
